### PR TITLE
Add note and image attachment support in payment

### PR DESCRIPTION
**Pull request by cp-nirali-s**
https://github.com/user-attachments/assets/0c8a2023-d8fd-4a16-a9ae-fd79653e54fa

## Release Notes

- **New Features**
  - Enhanced color definitions for better UI appearance.
  - Introduced new properties in expense and transaction models to track updates and additional metadata.
  - Added image handling capabilities in payment and transaction management.
  - Improved handling of image URLs in group management.
  - Added functionality for managing notes and images in group payments.
  - Introduced new state variables and streamlined UI components for better image and note management in transaction views.

- **Improvements**
  - Streamlined methods for updating and deleting expenses and transactions, ensuring better data integrity.
  - Improved localization strings for user interactions.
  - Enhanced image processing during group creation and updates.
  - Improved user interface and interaction flow in transaction detail views.
  - Simplified logic for displaying image picker options.

- **Refactor**
  - Renamed and reorganized several view components for clarity and consistency.
  - Updated various view models to enhance state management and user experience.
  - Removed redundant properties and methods to streamline code.

- **Bug Fixes**
  - Fixed issues related to image display and note handling in transaction views.
  - Updated logic for displaying the "Remove" button in image picker options.
  - Adjusted image display logic and note handling in group payment views.

### CODE REVIEW

File: Splito/UI/Home/Expense/AddExpenseView.swift [hunk 1]
```diff
@@ -34,10 +34,9 @@ struct AddExpenseView: View {
                 .scrollIndicators(.hidden)
                 .scrollBounceBehavior(.basedOnSize)
 
-                AddExpenseFooterView(date: $viewModel.expenseDate, showImagePickerOptions: $viewModel.showImagePickerOptions,
-                                     expenseImage: viewModel.expenseImage, expenseImageUrl: viewModel.expenseImageUrl,
-                                     handleNoteBtnTap: viewModel.handleNoteBtnTap, handleExpenseImageTap: viewModel.handleExpenseImageTap,
-                                     handleActionSelection: viewModel.handleActionSelection(_:))
+                FooterView(date: $viewModel.expenseDate, showImagePickerOptions: $viewModel.showImagePickerOptions,
```
Comment (cp-amisha-i) [inline Splito/UI/Home/Expense/AddExpenseView.swift:37]: Let's add some prefixe to this view name, direct `FooterView` is not that much meaningful.

File: Splito/UI/Home/Expense/AddExpenseView.swift [hunk 2]
```diff
@@ -224,16 +221,16 @@ private struct AddExpenseFooterView: View {
         HStack(spacing: 16) {
             Spacer()
 
-            DatePickerView(date: $date, isForAddExpense: true)
+            DatePickerView(date: $date)
 
-            ExpenseImagePickerView(image: expenseImage, imageUrl: expenseImageUrl, handleImageBtnTap: handleExpenseImageTap)
+            ExpenseImagePickerView(image: image, imageUrl: imageUrl, handleImageBtnTap: handleImageTap)
```
Comment (cp-amisha-i) [inline Splito/UI/Home/Expense/AddExpenseView.swift:226]: What if we just keep `ImagePickerView`?

Comment (cp-nirali-s) [reply Splito/UI/Home/Expense/AddExpenseView.swift:226]: We already have ImagePickerView struct

Comment (cp-amisha-i) [reply Splito/UI/Home/Expense/AddExpenseView.swift:226]: Then why we are not using it?

Comment (cp-nirali-s) [reply Splito/UI/Home/Expense/AddExpenseView.swift:226]: We are using it and below is ImagePickerView
simulator_screenshot_A13218C0-0AD5-478C-BEC6-12350B53A435

**Pull request merged**